functions/public/Invoke-WPFUIElements.ps1

```diff
@@ -0,0 +1,315 @@
+function Invoke-WPFUIElements {
+    <#
+    .SYNOPSIS
+        Adds UI elements to a specified Grid in the WinUtil GUI based on a JSON configuration.
+    .PARAMETER configVariable
+        The variable/link containing the JSON configuration.
+    .PARAMETER targetGridName
+        The name of the grid to which the UI elements should be added.
+    .PARAMETER columncount
+        The number of columns to be used in the Grid. If not provided, a default value is used based on the panel.
+    .EXAMPLE
+        Invoke-WPFUIElements -configVariable $sync.configs.applications -targetGridName "install" -columncount 5
+    .NOTES
+        Future me/contributer: If possible please wrap this into a runspace to make it load all panels at the same time.
+    #>
+
+    param(
+        [Parameter(Mandatory)]
+        [PSCustomObject]$configVariable,
+
+        [Parameter(Mandatory)]
+        [string]$targetGridName,
+
+        [Parameter(Mandatory)]
+        [int]$columncount
+    )
+
+    $theme = $sync.configs.themes.$ctttheme
+
+    # Convert PSCustomObject to Hashtable
+    $configHashtable = @{}
+    $configVariable.PSObject.Properties.Name | ForEach-Object {
+        $configHashtable[$_] = $configVariable.$_
+    }
+
+    $organizedData = @{}
+    # Iterate through JSON data and organize by panel and category
+    foreach ($entry in $configHashtable.Keys) {
+        $entryInfo = $configHashtable[$entry]
+
+        # Create an object for the application
+        $entryObject = [PSCustomObject]@{
+            Name = $entry
+            Order = $entryInfo.order
+            Category = $entryInfo.Category
+            Content = $entryInfo.Content
+            Choco = $entryInfo.choco
+            Winget = $entryInfo.winget
+            Panel = if ($entryInfo.Panel) { $entryInfo.Panel } else { "0" }
+            Link = $entryInfo.link
+            Description = $entryInfo.description
+            Type = $entryInfo.type
+            ComboItems = $entryInfo.ComboItems
+            Checked = $entryInfo.Checked
+            ButtonWidth = $entryInfo.ButtonWidth
+        }
+
+        if (-not $organizedData.ContainsKey($entryObject.Panel)) {
+            $organizedData[$entryObject.Panel] = @{}
+        }
+
+        if (-not $organizedData[$entryObject.Panel].ContainsKey($entryObject.Category)) {
+            $organizedData[$entryObject.Panel][$entryObject.Category] = @()
+        }
+
+        # Store application data in an array under the category
+        $organizedData[$entryObject.Panel][$entryObject.Category] += $entryObject
+    }
+
+    # Retrieve the main window and the target Grid by name
+    $window = $sync["Form"]
+    $targetGrid = $window.FindName($targetGridName)
+
```

Comment:
```suggestion

    $BorderStyle = $window.FindResource("BorderStyle")
    $ColorfulToggleSwitchStyle = $window.FindResource("ColorfulToggleSwitchStyle")
    $HoverTextBlockStyle = $window.FindResource("HoverTextBlockStyle")

```
@MyDrift-user The above snippet of code is a suggestion on caching the result of `FindResource` Method, which's done at least once or twice for every entry iteration, which may or may not hurt performance by a marginable amount. Of course you'll need to replace the `FindResource` calls in this script with these cached results.. and I suggest Error Checking it, you can do a simple if statement, like so:
```ps1
if ($BorderStyle -eq $null -or $...) {
    throw [GenericException]::new("Failed to retrieve Styles using 'FindResource' from main window element.")
}
```
or through other ways if you choose so (preferably ones that provide exactly which style the script failed to retrieve.. instead of just "Oops, couldn't find the resource, bye." 😅)

Reply:
done